packages/services/src/serverconnection.ts

```diff
@@ -47,328 +107,137 @@ namespace ServerConnection {
+   * The default settings.
    */
   export
-  interface IError {
-    /**
-     * The event triggering the error.
-     */
-    readonly event: Event;
-
-    /**
-     * The XHR object.
-     */
-    readonly xhr: XMLHttpRequest;
-
-    /**
-     * The request input data.
-     */
-    readonly request: IRequest;
-
-    /**
-     * The settings associated with the request.
-     */
-    readonly settings: ISettings;
-
-    /**
-     * The error message associated with the error.
-     */
-    readonly message: string;
-  }
+  const defaultSettings: ServerConnection.ISettings = {
+    baseUrl: PageConfig.getBaseUrl(),
+    wsUrl: PageConfig.getWsUrl(),
+    token: PageConfig.getToken(),
+    init: { 'cache': 'no-cache', 'credentials': 'same-origin' },
+    fetch: FETCH,
+    Headers: HEADERS,
+    Request: REQUEST,
+    WebSocket: WEBSOCKET
+  };
 }
 
 
 /**
  * The namespace for module private data.
  */
 namespace Private {
-  export
-  let defaultSettings: ServerConnection.ISettings;
-
   /**
    * Handle the server connection settings, returning a new value.
    */
   export
   function makeSettings(options: Partial<ServerConnection.ISettings> = {}): ServerConnection.ISettings {
-    let baseUrl = options.baseUrl || PageConfig.getBaseUrl();
     return {
-      baseUrl,
-      wsUrl: options.wsUrl || PageConfig.getWsUrl(baseUrl),
-      user: options.user || '',
-      password: options.password || '',
-      withCredentials: !!options.withCredentials,
-      timeout: options.timeout || 0,
-      token: options.token || PageConfig.getOption('token'),
-      requestHeaders: { ...options.requestHeaders || {} },
-      xhrFactory: options.xhrFactory || xhrFactory,
-      wsFactory: options.wsFactory || wsFactory
+      ...ServerConnection.defaultSettings,
+      ...options
     };
   }
 
   /**
-   * Make an xhr request using settings.
+   * Handle a request.
    */
   export
-  function populateRequest(xhr: XMLHttpRequest, request: ServerConnection.IRequest, settings: ServerConnection.ISettings): void {
-    if (request.contentType !== void 0) {
-      xhr.setRequestHeader('Content-Type', request.contentType);
-    }
-
-    xhr.timeout = settings.timeout;
-    if (settings.withCredentials) {
-      xhr.withCredentials = true;
-    }
+  function handleRequest(url: string, init: RequestInit, settings: ServerConnection.ISettings): Promise<Response> {
 
-    // Write the request headers.
-    let headers = request.headers;
-    if (headers) {
-      for (let prop in headers) {
-        xhr.setRequestHeader(prop, headers[prop]);
-      }
+    // Handle notebook server requests.
+    if (url.indexOf(settings.baseUrl) !== 0) {
+      throw new Error('Can only be used for notebook server requests');
     }
 
-    headers = settings.requestHeaders;
-    for (let prop in headers) {
-      xhr.setRequestHeader(prop, headers[prop]);
-    }
+    let request = new settings.Request(url, init);
+    request = new settings.Request(request, settings.init);
```

Comment:
This is not a pattern I could find anywhere else, so I had to check the specs for what would happen. AFAICT, the settings in `init` will be overridden by `settings.init` here, and should (?) be equivalent to doing
```ts
let request = new settings.Request(url, {...init, ...settings.init});
```
Is this what was intended? I would maybe expect the individual request's init to take precedence. It should probably be commented/documented somehow.